### PR TITLE
feat: add `blas/base/zcopy`

### PR DESCRIPTION
**Pull request by aman-095**
## Description

> What is the purpose of this pull request?

This RFC proposes to add a routine to copy values from one complex double-precision floating-point vector to another complex double-precision floating-point vector as defined in [BLAS Level 1](http://www.netlib.org/blas/#_level_1) routines. Specifically adding `@stdlib/blas/base/zcopy` is proposed.

## Related Issues

> Does this pull request have any related issues?
None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/zcopy/README.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+var zcopy = require( '@stdlib/blas/base/zcopy' );
+```
+
+#### zcopy( N, x, strideX, y, strideY )
+
+Copies values from `x` into `y`.
+
+```javascript
+var Complex128Array = require( '@stdlib/array/complex128' );
+var realf = require( '@stdlib/complex/realf' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/README.md]: The use of `realf`and `imagf` here and elsewhere in this PR is incorrect. You should be using `@stdlib/complex/real` which is for extracting the real component of a **double-precision floating-point number**.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/zcopy/README.md]: The `f` suffix tells you that the API is for single-precision floating-point numbers, not double-precision.

File: lib/node_modules/@stdlib/blas/base/zcopy/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        Output array.
+
+    Examples
+    --------
+    // Standard usage:
+    > var x = new {{alias:@stdlib/array/complex128}}( [ 1.0, 2.0, 3.0, 4.0 ] );
+    > var y = new {{alias:@stdlib/array/complex128}}( [ 6.0, 7.0, 8.0, 9.0 ] );
+    > {{alias}}( x.length, x, 1, y, 1 );
+    > var z = y.get( 0 );
+    > var re = {{alias:@stdlib/complex/realf}}( z )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/docs/repl.txt]: Same comment.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks, @aman-095. This looks good apart from the `realf` and `imagf` usage. Once resolved, this should be ready for merge.

**Comment by kgryte** [comment]
@aman-095 Are you planning on adding the C and Fortran implementations in a follow-up PR? Or will you be adding them to this PR?

**Comment by aman-095** [comment]
@kgryte I'll correct the implementation and yes I will add the C and Fortran implementation in this PR itself.

**Comment by kgryte** [comment]
@aman-095 Great! Thanks!

File: lib/node_modules/@stdlib/blas/base/zcopy/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	var f;
+	var i;
+
+	min = 1; // 10^min
+	max = 6; // 10^max
+
+	for ( i = min; i <= max; i++ ) {
+		len = pow( 10, i );
+		f = createBenchmark( len );
+		bench( pkg+':len='+len, opts, f );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/benchmark/benchmark.native.js]:
```suggestion
		bench( pkg+'::native:len='+len, opts, f );
```

Missing the benchmark namespace.

File: lib/node_modules/@stdlib/blas/base/zcopy/benchmark/benchmark.ndarray.native.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	var f;
+	var i;
+
+	min = 1; // 10^min
+	max = 6; // 10^max
+
+	for ( i = min; i <= max; i++ ) {
+		len = pow( 10, i );
+		f = createBenchmark( len );
+		bench( pkg+':ndarray:len='+len, opts, f );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/benchmark/benchmark.ndarray.native.js]:
```suggestion
		bench( pkg+'::native:ndarray:len='+len, opts, f );
```

File: lib/node_modules/@stdlib/blas/base/zcopy/benchmark/fortran/benchmark.length.f [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        call print_results( iter, elapsed )
+        ! ..
+        write (str, '(I15)') count
+        tmp = adjustl( str )
+        print '(A,A,A)', 'ok ', trim( tmp ), ' benchmark finished'
+      end do
+    end do
+    call print_summary( count, count )
+  end subroutine main
+end program bench
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/benchmark/fortran/benchmark.length.f]:
```suggestion
end program bench

```

File: lib/node_modules/@stdlib/blas/base/zcopy/benchmark/fortran/benchmark.length.f [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    ! Intrinsic functions:
+    intrinsic random_number, cpu_time, cmplx
+    ! ..
+    ! Allocate arrays:
+    allocate( x(len), y(len) )
+    ! ..
+    do i = 1, len
+      call random_number( r1 )
+      call random_number( r2 )
+      x( i ) = cmplx( (real(r1)*10000.0d0)-5000.0d0, (real(r2)*1000.0d0)-5000.0d0, kind=8  )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/benchmark/fortran/benchmark.length.f]:
```suggestion
      x( i ) = cmplx( (r1*10000.0d0)-5000.0d0, (r2*1000.0d0)-5000.0d0, kind=kind(0.0d0) )
```

`r1` and `r2` are double-precision. Not clear why you are converting to single-precision here. And use `kind()` here, as using the literal `8` is not universally portable.

File: lib/node_modules/@stdlib/blas/base/zcopy/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,132 @@
+
+{{alias}}( N, x, strideX, y, strideY )
+    Copies values from one complex double-precision floating-point vector to
+    another complex double-precision floating-point vector.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/docs/repl.txt:4]:
```suggestion
    Copies values from one double-precision complex floating-point vector to
    another double-precision complex floating-point vector.
```

File: lib/node_modules/@stdlib/blas/base/zcopy/lib/ndarray.native.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex128' );
+var minViewBufferIndex = require( '@stdlib/strided/base/min-view-buffer-index' );
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Copies values from one complex single-precision floating-point vector to another complex single-precision floating-point vector.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/lib/ndarray.native.js]:
```suggestion
* Copies values from one complex double-precision floating-point vector to another complex double-precision floating-point vector.
```

File: lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+! ## Authors
+!
+! * Univ. of Tennessee
+! * Univ. of California Berkeley
+! * Univ. of Colorado Denver
+! * NAG Ltd.
+!
+! ## History
+!
+! * Jack Dongarra, linpack, 3/11/78.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f]:
```suggestion
! * Jack Dongarra, linpack, 4/11/78.
```

The extended comment should match what is in netlib: https://github.com/OpenMathLib/OpenBLAS/blob/develop/lapack-netlib/BLAS/SRC/zcopy.f

File: lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+      iy = 1
+    end if
+    do i = 1, N
+      zy( iy ) = zx( ix )
+      ix = ix + strideX
+      iy = iy + strideY
+    end do
+  end if
+  return
+end subroutine zcopy
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f]:
```suggestion
end subroutine zcopy

```

File: lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+! @param {integer} strideY - `zy` stride length
+!<
+subroutine zcopy( N, zx, strideX, zy, strideY )
+  implicit none
+  ! ..
+  ! Scalar arguments:
+  integer :: strideX, strideY, N
+  ! ..
+  ! Array arguments:
+  complex(kind=8) :: zx(*), zy(*)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f]:
```suggestion
  complex(kind=kind(0.0d0) :: zx(*), zy(*)
```

File: lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+! > Like all software, it is copyrighted. It is not trademarked, but we do ask the following:
+! >
+! > * If you modify the source for these routines we ask that you change the name of the routine and comment the changes made to the original.
+! >
+! > * We will gladly answer any questions regarding the software. If a modification is done, however, it is the responsibility of the person who modified the routine to provide support.
+!
+! @param {integer} N - number of indexed elements
+! @param {Array<complex(kind=8)>} zx - input array
+! @param {integer} strideX - `zx` stride length
+! @param {Array<complex(kind=8)>} zy - output array
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f]:
```suggestion
! @param {Array<complex<double>>} zx - input array
! @param {integer} strideX - `zx` stride length
! @param {Array<complex<double>>} zy - output array
```

Fortran is a bit annoying in its parameterization of dtypes.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

File: lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+! @param {integer} strideY - `zy` stride length
+!<
+subroutine zcopy( N, zx, strideX, zy, strideY )
+  implicit none
+  ! ..
+  ! Scalar arguments:
+  integer :: strideX, strideY, N
+  ! ..
+  ! Array arguments:
+  complex(kind=kind(0.0d0) :: zx(*), zy(*)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/zcopy/src/zcopy.f]:
```suggestion
  complex(kind=kind(0.0d0)) :: zx(*), zy(*)
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**